src/main/java/de/tum/cit/aet/artemis/programming/service/ProgrammingExerciseImportFromFileService.java

```diff
@@ -170,44 +172,71 @@ private void importRepositoriesFromFile(ProgrammingExercise newExercise, Path ba
+        copyImportedExerciseContentToRepositories(templateRepo, solutionRepo, testRepo, auxiliaryRepositories, basePath);
+        replaceImportedExerciseShortName(Map.of(oldExerciseShortName, newExercise.getShortName()), List.of(solutionRepo, templateRepo, testRepo));
+        replaceImportedExerciseShortName(Map.of(oldExerciseShortName, newExercise.getShortName()), auxiliaryRepositories);
 
         gitService.stageAllChanges(templateRepo);
         gitService.stageAllChanges(solutionRepo);
         gitService.stageAllChanges(testRepo);
+        for (Repository auxRepo : auxiliaryRepositories) {
+            gitService.stageAllChanges(auxRepo);
+        }
 
         gitService.commitAndPush(templateRepo, "Import template from file", true, user);
         gitService.commitAndPush(solutionRepo, "Import solution from file", true, user);
         gitService.commitAndPush(testRepo, "Import tests from file", true, user);
+        for (Repository auxRepo : auxiliaryRepositories) {
+            gitService.commitAndPush(auxRepo, "Import auxiliary repo from file", true, user);
+        }
+
     }
 
-    private void replaceImportedExerciseShortName(Map<String, String> replacements, Repository... repositories) {
+    private void replaceImportedExerciseShortName(Map<String, String> replacements, List<Repository> repositories) {
         for (Repository repository : repositories) {
             fileService.replaceVariablesInFileRecursive(repository.getLocalPath(), replacements, SHORT_NAME_REPLACEMENT_EXCLUSIONS);
         }
     }
 
-    private void copyImportedExerciseContentToRepositories(Repository templateRepo, Repository solutionRepo, Repository testRepo, Path basePath) throws IOException {
+    private void copyImportedExerciseContentToRepositories(Repository templateRepo, Repository solutionRepo, Repository testRepo, List<Repository> auxiliaryRepositories,
+            Path basePath) throws IOException {
         repositoryService.deleteAllContentInRepository(templateRepo);
         repositoryService.deleteAllContentInRepository(solutionRepo);
         repositoryService.deleteAllContentInRepository(testRepo);
-        copyExerciseContentToRepository(templateRepo, RepositoryType.TEMPLATE, basePath);
-        copyExerciseContentToRepository(solutionRepo, RepositoryType.SOLUTION, basePath);
-        copyExerciseContentToRepository(testRepo, RepositoryType.TESTS, basePath);
+        for (Repository auxRepo : auxiliaryRepositories) {
+            repositoryService.deleteAllContentInRepository(auxRepo);
+        }
+
+        copyExerciseContentToRepository(templateRepo, RepositoryType.TEMPLATE.getName(), basePath);
+        copyExerciseContentToRepository(solutionRepo, RepositoryType.SOLUTION.getName(), basePath);
+        copyExerciseContentToRepository(testRepo, RepositoryType.TESTS.getName(), basePath);
+        for (Repository auxRepo : auxiliaryRepositories) {
+            String auxRepoSuffix = auxRepo.getLocalPath().toString().split("-")[1];
```

Comment:
Can aux repo names also contain `-`?

Reply:
good catch, they can and that would break the current logic